src/constants.py

```diff
@@ -6,6 +6,8 @@
 The use of this module should be limited to cases where the constant is not better placed in
 another module or to resolve circular imports.
 """
+DEFAULT_BRANCH = "main"
```

Comment:
Perhaps we could get the default branch exporting the GH event information into a env var

Reply:
As a matter of fact, we were thinking with @jdkandersson to address this in a separate PR, where we also re-introduce the e2e integration tests. Is it fine if we resolve this comment within the scope of this PR?

Reply:
We should refactor this anyway to ask the repo what the default branch is, probably ok to leave for this PR